src/main/scala/com/fulcrumgenomics/vcf/api/VcfSource.scala

```diff
@@ -0,0 +1,114 @@
+  * index is present.
+  *
+  * @param reader the underlying HTSJDK [[VCFFileReader]]
+  */
+class VcfSource private(private val reader: AbstractFeatureReader[VariantContext, _]) extends IterableView[Variant, VcfSource] with Closeable {
+  /** The header associated with the VCF being read. */
+  val header: VcfHeader = VcfConversions.toScalaHeader(reader.getHeader.asInstanceOf[VCFHeader])
+
+  /**
+    * The type of iterator returned by both the [[iterator]] method as well as the [[query()]] method. Note that
+    * [[SelfClosingIterator]] both self-closes when it hits the end of the iterator, _and_ extends
+    * [[com.fulcrumgenomics.commons.collection.BetterBufferedIterator]].
+    */
+  type VariantIterator = SelfClosingIterator[Variant]
+
+  /** Wraps an iterator provided by HTSJDK into a SelfClosingIterator that transforms VariatnContexts into Variants. */
+  private def wrap(it: CloseableIterator[VariantContext]): VariantIterator = {
+    new SelfClosingIterator(
+      iter   = it.map(vc => VcfConversions.toScalaVariant(vc, header)),
+      closer = () => it.close())
+  }
+
+  /**
+    * Returns an iterator over the entire stream of variants. The returned iterator may be be closed by invoking
+    * `close()` on it, and will automatically close itself when exhausted.  Only a single iterator at a time
+    * is supported per [[VcfSource]], including iterators returned from [[query()]].
+    */
+  override def iterator: VariantIterator = wrap(reader.iterator())
+
+  /** True if the VCF is sorted and indexed such that queries can be executed, false otherwise. */
+  def isQueryable: Boolean = this.reader.isQueryable
+
+  /**
+    * Returns an iterator over variants overlapping the specified genomic region.
+    *
+    * The returned iterator may be be closed by invoking `close()` on it, and will automatically close itself
+    * when exhausted.  Only a single iterator at a time is supported per [[com.fulcrumgenomics.vcf.api.VcfSource]], including iterators
+    * returned from [[iterator()]].
+    */
+  def query(chrom: String, start: Int, end: Int): Iterator[Variant] = wrap(reader.query(chrom, start, end))
+
+  /** Closes the underlying reader. */
+  override def close(): Unit = this.reader.safelyClose()
+
+  /** Required by [[scala.collection.IterableView]]. */
+  override protected def underlying: VcfSource = this
+}
+
+object VcfSource {
+  /**
+    * Manufactures a variant source for reading from the specified path.  The index, if one exists, will be
+    * auto-discovered based on the path to the VCF.
+    *
+    * @param path the path to a VCF, gzipped VCF or BCF file
+    * @return a VariantSource for reading from the path given
+    */
+  def apply(path: PathToVcf): VcfSource = {
+    val codec  = if (PathUtil.extensionOf(path).contains(".bcf")) {
+      new BCF2Codec
+    }
+    else {
+      val c = new VCFCodec
+      c.disableOnTheFlyModifications()
```

Comment:
This is the only way to stop HTSJDK borking up the headers to match standard GATK headers. So we end up having to use the tribble readers directly instead of `VCFFileReader`.  Sigh.

Reply:
awesome